content/docs/user-guide/troubleshooting.md

```diff
@@ -107,3 +107,20 @@ experiments.
 [git credentials]: https://git-scm.com/docs/gitcredentials
 [ssh git url]:
   https://git-scm.com/book/en/v2/Git-on-the-Server-The-Protocols#_the_protocols
+
+## Failed to open pickled cache {#pickle}
+
+You may encounter this error when using DVC with multiple Python versions within
```

Comment:
Can we generalize/simplify this a bit to avoid mentioning pickling? Prob. an unnecessary implementation detail

Please see https://github.com/iterative/dvc/pull/7300#pullrequestreview-864458242

Reply:
And prob. link to https://dvc.org/doc/user-guide/project-structure/internal-files somewhere here 🙂

Reply:
I'm not sure it's irrelevant here, because the error specifically happens due to using pickle in mixed python versions.

Reply:
Maybe explain the pickling issue so it's clear why it's mentioned? For example: "Pickle protocol version 5 (https://docs.python.org/3/library/pickle.html#data-stream-format) is used in Python 3.8+ but is not supported in earlier Python versions."

Also, maybe it's better to reverse the paragraph order so that the fix is explained up front and implementation details are explained later?

Reply:
> the error specifically happens due to using pickle in mixed python versions

But that's used internally in DVC right? That's what I mean by implementation detail.

Or is this about user code or data that uses pickling from different Python versions?

Reply:
UPDATE: OK I read the explanations in https://github.com/iterative/dvc/pull/7222#discussion_r789406572 and https://github.com/iterative/dvc/pull/7300#pullrequestreview-864458242 and I see it's an implementation detail indeed so I still incline to simplify this in some way that pickling isn't mentioned.

I'd be happy to make a specific suggestion or take this over but I still don't fully understand the problem... See my questions in https://github.com/iterative/dvc/pull/7300#discussion_r797215402.

Reply:
I still think including that the error is due to pickling is relevant, otherwise it's unclear why the error occurs in the first place. Since DVC officially supports both Python 3.8+ and 3.7, it is non-obvious why DVC would work in a given repo in 3.8+ but not 3.7.

Reply:
> I still think including that the error is due to pickling is relevant

Actually I just realized we never included that remark in the text, only in the section title. I would prefer to completely avoid it but since that's how it was merged in the core error message, I'll leave it here too.